GPII-1318: Updated to node 4.x-ready branch of universal

amb26 opened the pull request:
Native build now seems reliable so node_modules can be deleted

javihernandez commented:
@amb26 sweet!!

sgithens commented:
@amb26 
I believe the build steps have changed for this refactor correct?  Could you update the README to reflect the changes.

kaspermarkus commented on gpii/node_modules/registryResolver/index.js:
```diff
@@ -1,18 +1,16 @@
-});
\ No newline at end of file
+// Contribute the registration of the registryResolver into the LifecycleManager's configuration
+fluid.makeGradeLinkage("gpii.windows.resolverConfigWithRegistry", 
```
trailing whitespace breaking jshint

kaspermarkus commented on gpii/node_modules/spiSettingsHandler/test/testSpiSettingsHandler.js:
```diff
@@ -18,16 +18,16 @@ https://github.com/GPII/universal/blob/master/LICENSE.txt
 
 var testPayloads = ["testMouseTrails", "testFilterKeys", "testMouse", "testMouseClickLock", "testLogFont", "testNonClientMetrics", "testStickyKeys", "testHighContrast"];
-
+/*
```
Why are the tests here commented out? Also if they remain commented out, the "testPayloads" definition of line 29 should also be commented out

amb26 replied:
Pure blunder... I was just trying to do some light testing on my machine without destroying its desktop

kaspermarkus commented on .gitignore:
```diff
@@ -0,0 +1,11 @@
+/gpii/node_modules/gpii-oauth2/gpii-oauth2-utilities/node_modules/
+
+npm-debug.log
+*.bak
```
This should be fixed to match what we want ignored in windows

amb26 commented:
Updated and working with the dynamic device reporter with corresponding commits in https://github.com/GPII/universal/pull/425. Incorporated material from https://github.com/GPII/windows/pull/66/files with fixes, and fixed broken tests from https://github.com/GPII/windows/pull/59
